Add CRD upgrade to v1beta2 to the API conversion tool (#4482)
strimzi/strimzi-kafka-operator: api-conversion/README.md

                          Creates an output YAML file for the converted custom resource
```

> **Once you converted the YAML files, apply the changes to Kubernetes using `kubectl apply -f` or `kubectl replace -f`.** 

Reviewer: ```suggestion
**When you have converted the YAML files, apply the changes to Kubernetes using `kubectl apply -f` or `kubectl replace -f`.** 
```

Reviewer: Is the api-conversion.sh a _kubectl exec_?

Author: No, none of these is using `kubectl` directly. The `convert-resource` is using Kubernetes APIs directly. `convert-file` just outputs a file and the user has to deal with it (might be `kubectl apply`, but also just committing it to Git and waiting for some pipeline to deploy it etc.)

Reviewer: Okay. Thanks

Reviewer: I can't add comments to the convert-resource help lines. But, to make consistent with the convert-files:

Convert Custom Resources directly in Kubernetes
  -a, --all-namespaces   **Converts** resources in all namespaces
  -d, --debug            **Runs the tool in debug mode**
  -k, --kind[=<kinds> [<kinds> [<kinds> [<kinds> [<kinds> [<kinds> [<kinds>
        [<kinds> [<kinds> [<kinds>]]]]]]]]]]
                         **Specifies the kinds of custom resources to be converted, or converts all resources if not specified**
      -ll, --log-level=<level>
                         **Sets the log level to enable logging**
  -n, --namespace=<namespace>
                         **Specifies a Kubernetes namespace or OpenShift project, or uses the current namespace if not specified**
      --name=<name>      **If --namespace and a single custom resource --kind is used, specifies the name of the custom resource being converted**

  -k, --kind[=<kinds> [<kinds> [<kinds> [<kinds> [<kinds> [<kinds> [<kinds>
        [<kinds> [<kinds> [<kinds>]]]]]]]]]]
                         Specifies the kinds of custom resources to be
                           converted, or converts all resources if not specified
      -ll, --log-level=<level>
                         Sets the log level to enable logging
  -n, --namespace=<namespace>
                         Specifies a Kubernetes namespace or OpenShift project,
                           or uses the current namespace if not specified
      --name=<name>      Name of the resource which should be converted (can be
                           used onl with --namespace and single --kind options)
```

### Required access rights

To convert the resources directly in your Kubernetes cluster, you must run the tool as a user with RBAC permission to:

* Get the Strimzi custom resources you are going to convert (when using the `--name` option)
* List the Strimzi custom resources you are going to convert (when not using the `--name` option)
* Replace the Strimzi custom resources you are going to convert

## Upgrading CRDs to v1beta2

> You can also use the tool to update the Strimzi CRDs to use the new `v1eta2` version as a stored version.

Reviewer: ```suggestion
You can also use the tool to update the Strimzi CRDs to use the new `v1beta2` version as a stored version.
```

This is required before upgrading to Strimzi 0.23 and newer.
You have to first convert all Strimzi custom resources to `v1beta2` using one of the previous commands.
Once the custom resources are ready, you can use the `crd-upgrade` subcommand to upgrade the CRDs.

> The following example command show how the tool is used:

Reviewer: ```suggestion
The following example command shows how the tool is used:
```

      --name=<name>      Name of the resource which should be converted (can be

>                            used onl with --namespace and single --kind options)

Reviewer: ```suggestion
                           used onl with --namespace and single --kind options)
```

```

### Required access rights

> To convert the resources directly in your Kubernetes cluster, you need to run the tool with a user which has the following RBAC rights:

Reviewer: ```suggestion
To convert the resources directly in your Kubernetes cluster, you must run the tool as a user with the following RBAC rights:
```

* Get the Strimzi custom resources you are going to convert (when using the `--name` option)
* List the Strimzi custom resources you are going to convert (when not using the `--name` option)
* Replace the Strimzi custom resources you are going to convert

## Upgrading CRDs to v1beta2

You can also use the tool to update the Strimzi CRDs to use the new `v1eta2` version as a stored version.
This is required before upgrading to Strimzi 0.23 and newer.
You have to first convert all Strimzi custom resources to `v1beta2` using one of the previous commands.
Once the custom resources are ready, you can use the `crd-upgrade` subcommand to upgrade the CRDs.
The following example command show how the tool is used:

```
# Upgrade the Strimzi CRDs to v1beta2
> bin/api-conversion.sh crd-upgrade
```

> **Once you upgrade the CRDs to use `v1beta2` as storage version, you should only use the fields available in `v1beta2` in your custom resources.**

Reviewer: ```suggestion
**After you have upgraded the CRDs to use `v1beta2` as storage version, you must only use the fields available in `v1beta2` in your custom resources.**
```

### Required access rights

> To upgrade the Strimzi CRDs to `v1beta2`, you need to run the tool with a user which has the following RBAC rights:

Reviewer: ```suggestion
To upgrade the Strimzi CRDs to `v1beta2`, you must run the tool as a user with RBAC permission to:
```
